.github/workflows/ci.yml

```diff
@@ -41,6 +41,17 @@ jobs:
       - run: npm run lint:static-site
       - run: node ./scripts/check-resource-index-match.js
 
+  type-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version-file: 'package.json'
+      - run: npm ci
+      - run: npm run build
+      - run: npm run type-check
```

Comment:
if the type-checking depends on having a build, wouldn't it be better to include the build part in the `type-check` entry in `package.json:scripts`? 

IOW, make this only `npm run type-check` here and...